lib/mongo/collection/view/readable.rb

```diff
@@ -461,26 +475,18 @@ def query_spec
         end
 
         def read_pref_formatted
-          read.to_mongos
+          @read_formatted ||= read.to_mongos
         end
 
         def special_selector
-          SPECIAL_FIELDS.reduce({}) do |hash, (key, method)|
-            value = send(method) || (options[:modifiers] && options[:modifiers][key])
-            hash[key] = value unless value.nil?
-            hash
-          end
+          sel = { :$query => selector }.merge(modifiers)
```

Comment:
This should be:

```ruby
sel = BSON::Document.new(:$query => selector).merge(modifiers)
```

In order to get the indifferent access provided by a `BSON::Document` vs. a normal hash, also, the command monitoring spec guarantees the filter will also be a `BSON::Document`.